ecs/client.go

```diff
@@ -300,3 +306,43 @@ func (c *BasicECSClient) isNonRetryableErrorCode(code string) bool {
 		return false
 	}
 }
+
+// isTaskNotFoundError returns whether or not the error returned from ECS is
+// because the task cannot be found.
+func isTaskNotFoundError(err error) bool {
+	awsErr, ok := err.(awserr.Error)
+	if !ok {
+		return false
+	}
+	return awsErr.Code() == ecs.ErrCodeInvalidParameterException && strings.Contains(awsErr.Message(), "The referenced task was not found")
```

Comment:
you could just return `ok &&` and then the rest of the boolean if you prefer

Reply:
I also don't have strong feelings, but I think using fewer booleans on a single line reads more easily and makes it easier to reason about.

Reply:
Sure, fine with me